java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java

```diff
@@ -567,6 +567,22 @@ public static void remove(ContentProjectFilter filter) {
         INSTANCE.removeObject(filter);
     }
 
+    /**
+     * Set all BUILDING {@link EnvironmentTarget}s to FAILED state.
+     *
+     * @return the number of updated targets
+     */
+    public static int failStaleTargets() {
+        List<EnvironmentTarget> targets = HibernateFactory.getSession()
```

Comment:
How many of these objects to we expect in worst case?

If it is more than, say, 100, I would suggest an UPDATE query instead of SELECTing, hydrating, updating and saving back (I would imagine at a later point in time).

Reply:
In typical scenarios, it shouldn't be more than tens of objects.

But your suggestion sounds like a most natural approach and I like it. Let me change it.